cnf-certification-test/lifecycle/scaling/statefulset_scaling.go

```diff
@@ -0,0 +1,192 @@
+}
+
+func scaleStateFulsetHelper(clients *clientsholder.ClientsHolder, ssClient v1.StatefulSetInterface, statefulset *v1app.StatefulSet, replicas int32, timeout time.Duration, up bool) bool {
+	if up {
+		logrus.Trace("scale UP statefulset to ", replicas, " replicas ")
+	} else {
+		logrus.Trace("scale DOWN statefulset to ", replicas, " replicas ")
+	}
+
+	name := statefulset.Name
+	namespace := statefulset.Namespace
+
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Retrieve the latest version of statefulset before attempting update
+		// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+		ss, err := ssClient.Get(context.TODO(), name, v1machinery.GetOptions{})
+		if err != nil {
+			tnf.ClaimFilePrintf("failed to get latest version of statefulset %s:%s with error %s", namespace, name, err)
+			logrus.Error("failed to get latest version of statefulset ", namespace, ":", name)
+			return err
+		}
+		ss.Spec.Replicas = &replicas
+		_, err = clients.AppsClients.StatefulSets(namespace).Update(context.TODO(), ss, v1machinery.UpdateOptions{})
+		if err != nil {
+			logrus.Error("can't update statefulset ", namespace, ":", name)
+			return err
+		}
+		if !podsets.WaitForStatefulSetReady(namespace, name, timeout) {
+			logrus.Error("can't update statefulset ", namespace, ":", name)
+			return errors.New("can't update statefulset")
+		}
+		return nil
+	})
+	if retryErr != nil {
+		logrus.Error("can't scale statefulset ", namespace, ":", name, " error=", retryErr)
+		return false
+	}
+	return true
+}
+
+func TestScaleHpaStatefulSet(statefulset *v1app.StatefulSet, hpa *v1autoscaling.HorizontalPodAutoscaler, timeout time.Duration) bool {
+	clients := clientsholder.GetClientsHolder()
+	hpaName := hpa.Name
+	name, namespace := statefulset.Name, statefulset.Namespace
+	hpscaler := clients.K8sClient.AutoscalingV1().HorizontalPodAutoscalers(namespace)
+	min := int32(1)
+	if hpa.Spec.MinReplicas != nil {
+		min = *hpa.Spec.MinReplicas
+	}
+	max := hpa.Spec.MaxReplicas
+	if min <= 1 {
+		// scale up
+		min++
+		max++
+		scaleUp := true
+		pass := scaleHpaStatefulSetHelper(hpscaler, hpaName, name, namespace, min, max, timeout, scaleUp)
+		if !pass {
+			return false
+		}
+		// scale down
+		min--
+		max--
+		pass = scaleHpaStatefulSetHelper(hpscaler, hpaName, name, namespace, min, max, timeout, !scaleUp)
+		if !pass {
+			return false
+		}
+	} else {
+		// scale down
+		min--
+		max--
+		scaleUp := false
+		pass := scaleHpaStatefulSetHelper(hpscaler, hpaName, name, namespace, min, max, timeout, scaleUp)
+		if !pass {
+			return false
+		}
+		// scale up
+		min++
+		max++
+		pass = scaleHpaStatefulSetHelper(hpscaler, hpaName, name, namespace, min, max, timeout, !scaleUp)
+		if !pass {
+			return false
+		}
+	}
+	return true
+}
+
+func scaleHpaStatefulSetHelper(hpscaler hps.HorizontalPodAutoscalerInterface, hpaName, statefulsetName, namespace string, min, max int32, timeout time.Duration, up bool) bool {
```

Comment:
Minor suggestion: The "up" parameter for this function is only used for the "UP/DOWN" log trace at the beginning. I believe you should remove it and call logrus.Trace(up/down) right before calling this function.